0007-configuration-management-extension/README.md

```diff
@@ -0,0 +1,120 @@
+---
+name: Configuration Management Extension
+shortname: 7/CMX
+status: raw
+editor: Alexey Shmalko <ashmalko@cybervisiontech.com>
+---
+
+## Introduction
+
+The Configuration Management Extension is a [Kaa Protocol](/0001-kaa-protocol/README.md) extension.
+
+It is intended to manage endpoint configuration distribution.
+
+## Requirements and constraints
+### Problems and possible solutions
+
+1. _The server should know if a configuration has been applied by the endpoint._
+
+   Possible solutions:
+   - The client sends a publish message in a response to the new configuration provided by the server.
+
+     This means the server is initiating a request. That does not work well for HTTP, CoAP, and other protocols that only support client-initiated requests.
+
+   - Introduce `/applied` resource, so the client sends an additional request once configuration is applied.
+
+2. _An endpoint might be interested in a part of the general configuration only._
```

Comment:
I don't see a good use-case for this, and imho the complexity is too high. If you need several types of configuration, you can use different extension instances as per 1/KP.